src/test/java/edu/uw/edm/edmzuulproxy/filter/CertificateAuthenticationFilterTest.java

```diff
@@ -202,8 +208,69 @@ public void shouldAddEmptyAuthorizedProfilesHeaderIfNoneDefined() {
         assertTrue(header.isEmpty());
     }
 
+    @Test
+    public void shouldAddDocfinityHeaderToRequest() {
+        mockHttpServletRequest.setRequestURI("/docfinity/webservices/rest/metadata");
+        mockHttpServletRequest.setMethod("GET");
+        mockHttpServletRequest.addHeader(CERTIFICATE_NAME_HEADER, "mycert");
+
+        when(certificateAuthorizerService.isAllowedForUri(any(), any(), any(), any())).thenReturn(true);
+        when(certificateAuthorizerService.getAuthorizedProfilesForUri(any(), any())).thenReturn(Lists.newArrayList("testprofile1", "testprofile2"));
+
+        CertificateAuthenticationFilter filter = newFilter();
+        filter.run();
+
+        final String authHeader = RequestContext.getCurrentContext().getZuulRequestHeaders().get("authorization");
+        final String tokenHeader = RequestContext.getCurrentContext().getZuulRequestHeaders().get("x-xsrf-token");
+        final String cookieHeader = RequestContext.getCurrentContext().getZuulRequestHeaders().get("cookie");
+        assertThat(authHeader, is("Bearer apikey"));
+        assertThat(tokenHeader, is("edm-token"));
+        assertThat(cookieHeader, is("xsrf-token=edm-token"));
+    }
+
+
+    @Test
+    public void shouldNotAddDocfinityHeaderToRequestForCert() {
+        mockHttpServletRequest.setRequestURI("/docfinity/webservices/rest/metadata");
+        mockHttpServletRequest.setMethod("GET");
+        mockHttpServletRequest.addHeader(CERTIFICATE_NAME_HEADER, "mycert1");
```

Comment:
Consider renaming mycert1 to "unknownCert" so that it is clearer to reader the intent.

Reply:
renamed.